### ISSUE TITLE
[Feature Request] Allow to change the domain path via API

### ISSUE THREAD

**Issue by envoyr** [Froxlor/Froxlor#1000]
**Is your feature request related to a problem? Please describe.**
Admins without "Can change server settings" can't change the document root / path in the API but users in dashboard can do.

**Describe the solution you'd like**
It would be nice if an admin *without* "Can change server settings" can change the path (relative to customer-docroot) **via API** like the user can in the Dashboard:

![scrnli_27-11-2021_15-23-00](https://user-images.githubusercontent.com/81368729/143685383-8adc5c30-3377-42ac-92a7-616a752c66df.png)

A new Parameter like `path` in the Domains API would be good 😊

**Comment by d00p** [comment]
I think you are talking about the parameter `documentroot` (when using _Domains.add()_ or _Domains.update()_), so there would be no new parameter like path...it's just a permission based problem.
And you are correct, I'm unsure why it is required for the admin to have the `change_serversettings` permission to update the paths, this could originate from the fact that you *could* set a path outside the customer homedirectory to anywhere on the server as admin (see description below). I'll dig into that ...thanks

```
	 * @param string $documentroot
	 *        	optional, specify homedir of domain by specifying a directory (relative to customer-docroot), be aware, if path starts with / it it considered a full path, not relative to customer-docroot. Also specifying a URL is possible here (redirect), default empty (autogenerated)
```

**Comment by d00p** [comment]
It would be very helpful if you could also test this with your api-call/script to see if the solution is working as you intended

**Comment by envoyr** [comment]
Mhh, i get following responses for a reseller with change_serversettings = 0 when update `documentroot` for a customer `demo`:

✅ `/var/customers/webs/random/rrrr/` => 400 response, error as expected
✅ `/some/random/path/` => 400 response, error as expected
❌ `/some/random/webs/demo/example/` => 200 response, but no update
❌ `subfolder/` => 200 response, but no update
❌ `https://example.com` => 200 response, but no update

**Comment by envoyr** [comment]
Issue was fixed in [149c093](https://github.com/Froxlor/Froxlor/commit/149c0935fa0b32c1c484f328a1a7700a6f7c59fe) everything works now 😊
